### ISSUE TITLE
Treemacs consuming 100% CPU / Treemacs is querying texlab unnecessarily

### ISSUE THREAD

**Issue by frgomes** [Alexander-Miller/treemacs#747]
Apparently, treemacs queries all LSP components it knows of, even when not necessary. I said "apparently" since my investigation was superficial. More details:

## Related
https://github.com/frgomes/.emacs.d/issues/9

## Troubleshooting

I've observed 100% CPU utilization caused by treemacs.
From my very very limited knowledge of treemacs, I've turned off a couple of parameters but the problem persisted.

I was coding in Rust, and I've decided to renamed **rls** (the LSP component for Rust) so that treemacs would not find it. The problem persisted.

Then I've tried something which was in the back of my head since the morning: I'm using openSUSE Tumbleweed, which performed a rolling update containing 1344 texlive packages. I don't need texlive at all and I was thinking about removing them. So, as a sake of test, I've first renamed **textlab** (the LSP component for LaTeX) so that treemacs would not be able to find it.

This solved the issue.

## About this issue

So, why treemacs queries **texlab** (for symbols related to LaTeX) when I'm coding in Rust and only querying **rls** would be theoretically sufficient? Am I mistaken about that? 

Bug or a feature?

**Comment by frgomes** [comment]
Complementing the initial report:

I've observed that, even removing all texlive packages, the slowness persist.
I've then renamed texlab so that treemacs cannot find it. The the issue vanishes once for all.

I'm not able to tell if the root cause of slowness is treemacs or texlab.

**Comment by Alexander-Miller** [comment]
> So, why treemacs queries texlab (for symbols related to LaTeX) when I'm coding in Rust and only querying rls would be theoretically sufficient? Am I mistaken about that?

Treemacs doesn't do that. At the most treemacs can ask for a buffer's symbol list, but that must be done manually and uses the builtin `imenu` feature. The problem must be somewhere in the connection with the language server. And that is not something I can help you with on my own. Treemacs' connection with lsp-mode is that treemacs provides a generic extensions api that is used by lsp-treemacs to display features like hierarchical symbol trees. 

In other words: treemacs gets told what to do, but the process of finding, sorting and filtering items for treemacs to show, and deciding when and how often to do so, lies in the hands of lsp-treemacs. I suggest raising the issue over there as well and putting me on CC.

In addition there's 2 important pieces of information that will be very useful for solving this:

 - Have a look at the builtin profiler, it'lll tell you where the CPU spends most of its time
 - Kill emacs with SIGUSR2 if you're on linux. That will make Emacs stop in its tracks and show you a stack trace of what is was up to.
